### PR TITLE
WF-63881- Adding UG documentation to convert Word document to Markdown file

### PR DESCRIPTION
**Pull request by chrispojunior**
<table>
<tbody>
<tr>
<td>Task Link</td>
<td>https://syncfusion.atlassian.net/browse/WF-63881</td>
</tr>
<tr>
<td>Description</td>
<td>Add UG document for&nbsp;Word to Markdown Conversion</td>
</tr>
<tr>
<td>Solution</td>
<td>Added code snippet in&nbsp;Word to Markdown conversion platforms</td>
</tr>
<tr>
<td>Content review confirmation</td>
<td>Yes</td>
</tr>
</tbody>
</table>

### CODE REVIEW

File: File-Formats/DocIO/word-to-markdown.md [hunk 1]
```diff
@@ -0,0 +1,800 @@
...
+title: Word Document to Markdown conversion | DocIO | Syncfusion
+description: This section illustrates how to convert Word document to Markdown using Syncfusion Word library (Essential DocIO)
+platform: file-formats
+control: DocIO
+documentation: UG
+---
+
+# Word to Markdown Conversion
+
+## Markdown
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/word-to-markdown.md:11]: Remove this heading

File: File-Formats/DocIO/word-to-markdown.md [hunk 2]
```diff
@@ -0,0 +1,800 @@
...
+    }
+//Please download the helper files from the below link to save the stream as file and open the file for viewing in Xamarin platform
+//https://help.syncfusion.com/file-formats/docio/create-word-document-in-xamarin#helper-files-for-xamarin
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Supported Markdown elements
+
+The following table illustrates the supported Markdown elements in .NET Word (DocIO) library and how to set that Markdown elements in Word document.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/word-to-markdown.md:127]: The following table illustrates the supported Markdown elements in Word to Markdown conversion and how to set that Markdown elements in input Word document.

File: File-Formats/DocIO/word-to-markdown.md [hunk 3]
```diff
@@ -0,0 +1,800 @@
...
+                args.ImageStream.CopyTo(fileStreamOutput);
+            //Set the uri to be used for image in output markdown. 
+            args.Uri = imagepath;
+        }
+{% endhighlight %}
+{% endtabs %}
+
+## Supported Word document elements
+
+The following table shows the list of supported elements while converting from Word document to Markdown.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/word-to-markdown.md:684]: The following table shows the list of Word document elements supported in Word to Markdown conversion.

File: File-Formats/DocIO/word-to-markdown.md [hunk 4]
```diff
@@ -0,0 +1,800 @@
...
+</tr>
+<tr>
+<td style="width: 318.625px;">Hidden</td>
+<td style="width: 424.375px;">Considered as comments in Markdown syntax</td>
+</tr>
+<tr>
+<td style="width: 318.625px;">List</td>
+<td style="width: 424.375px;">Numbered and bulleted lists are supported.</td>
+</tr>
+</tbody>
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/word-to-markdown.md:799]: Add Limitations

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: File-Formats/DocIO/word-to-markdown.md (hunk 4 above)
Comment (chrispojunior) [reply File-Formats/DocIO/word-to-markdown.md:796]: I maintained each limitation in each area of occurrence.

File: File-Formats/DocIO/word-to-markdown.md [hunk 5]
```diff
@@ -0,0 +1,799 @@
...
+The following code example illustrates how to save Image files during Word to Markdown Conversion.
+
+{% tabs %}
+{% highlight c# tabtitle="C#" %}
+//Open an existing Word document. 
+using (WordDocument document = new WordDocument(@"Input.docx"))
+{
+    //Hook the event to customize the image. 
+    document.SaveOptions.ImageNodeVisited += SaveImage;
+    //Save the Word document as a Markdown file.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/word-to-markdown.md:554]: //Save a Word document as a Markdown file.

File: File-Formats/DocIO/word-to-markdown.md [hunk 6]
```diff
@@ -0,0 +1,799 @@
...
+    document.Save("WordtoMd.md", FormatType.Markdown);
+}
+The following code examples shows the event handler to customize the image path and save the image in external folder.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/word-to-markdown.md:557]: The following code examples show the event handler to customize the image path and save the image in an external folder.

File: File-Formats/DocIO/word-to-markdown.md [hunk 7]
```diff
@@ -0,0 +1,799 @@
...
+
+static void SaveImage(object sender, ImageNodeVisitedEventArgs args)
+        {
+            string imagepath = @"D:\Temp\Image1.png";
+            //Save the image stream as file.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/word-to-markdown.md:562]: //Save the image stream as a file.

File: File-Formats/DocIO/word-to-markdown.md [hunk 8]
```diff
@@ -0,0 +1,799 @@
...
+            args.Uri = imagepath;
+        }
+{% endhighlight %}
+
+{% highlight vb.net tabtitle="VB.NET" %}
+'Open an existing Word document. 
+Using document As WordDocument = New WordDocument("Input.docx")
+ 'Hook the event to customize the image. 
+document.SaveOptions.ImageNodeVisited += SaveImage
+ 'Save the Word document as a Markdown file.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/word-to-markdown.md:575]: 'Save a Word document as a Markdown file.

File: File-Formats/DocIO/word-to-markdown.md [hunk 9]
```diff
@@ -0,0 +1,799 @@
...
+document.Save("WordtoMd.md", FormatType.Markdown)
+End Using
+
+Private Shared Sub SaveImage(ByVal sender As Object, ByVal args As ImageNodeVisitedEventArgs)
+Dim imagepath = "D:\Temp\Image1.png"
+ 'Save the image stream as file. 
+Using fileStreamOutput = File.Create(imagepath)
+   args.ImageStream.CopyTo(fileStreamOutput)
+End Using
+'Set the uri to be used for image in output markdown.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/word-to-markdown.md:585]: 'Set the URI to be used for the image in the output markdown.

File: File-Formats/DocIO/word-to-markdown.md [hunk 10]
```diff
@@ -0,0 +1,799 @@
...
+        //Save the stream as a Markdown file in the local machine.
+        Save(stream, "WordtoMd.md");
+    }
+//Please refer the below link to save Markdown file in UWP platform
+//https://help.syncfusion.com/file-formats/docio/create-word-document-in-uwp#save-word-document-in-uwp
+}
+static void SaveImage(object sender, ImageNodeVisitedEventArgs args)
+        {
+            string imagepath = @"D:\Temp\Image1.png";
+            //Save the image stream as file.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/word-to-markdown.md:611]: //Save the image stream as a file.

File: File-Formats/DocIO/word-to-markdown.md [hunk 11]
```diff
@@ -0,0 +1,799 @@
...
+//Please refer the below link to save Markdown file in UWP platform
+//https://help.syncfusion.com/file-formats/docio/create-word-document-in-uwp#save-word-document-in-uwp
+}
+static void SaveImage(object sender, ImageNodeVisitedEventArgs args)
+        {
+            string imagepath = @"D:\Temp\Image1.png";
+            //Save the image stream as file. 
+            using (FileStream fileStreamOutput = File.Create(imagepath))
+                args.ImageStream.CopyTo(fileStreamOutput);
+            //Set the uri to be used for image in output markdown.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/word-to-markdown.md:614]: //Set the URI to be used for the image in the output markdown.

File: File-Formats/DocIO/word-to-markdown.md [hunk 12]
```diff
@@ -0,0 +1,799 @@
...
+        outStream.Dispose();
+    }
+}
+static void SaveImage(object sender, ImageNodeVisitedEventArgs args)
+        {
+            string imagepath = @"D:\Temp\Image1.png";
+            //Save the image stream as file. 
+            using (FileStream fileStreamOutput = File.Create(imagepath))
+                args.ImageStream.CopyTo(fileStreamOutput);
+            //Set the uri to be used for image in output Markdown.
```
Comment (Chrispineagunja) [inline File-Formats/DocIO/word-to-markdown.md:645]: //Set the URI to be used for the image in the output Markdown.

**Review by MohanaselvamJothi — APPROVED** [review]

**Pull request merged**